### PR TITLE
compact TUI to monitor layers download progress

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
As terminal height might not allow to display per-layers download progress status, render a compact TUI using 8-dots braille chars as progress status

see demo on https://asciinema.org/a/4jZWUZfhcXouJPd1p6h0yMxtC

### CODE REVIEW

**Review by laurazard — APPROVED** [review]
Nice!! Looks really good 🥳

**Comment by williamdes** [comment]
It's so cute!!
can we have a percentage right after the text?

**Comment by ndeloof** [comment]
this unfortunately breaks the progress alignment, need to double check

File: pkg/compose/push.go [hunk 1]
```diff
@@ -139,10 +139,12 @@ func toPushProgressEvent(prefix string, jm jsonmessage.JSONMessage, w progress.W
...
 	var (
-		text   string
-		status = progress.Working
+		text    string
+		status  = progress.Working
+		total   int64
+		current int64
 	)
-	if jm.Status == "Pull complete" || jm.Status == "Already exists" {
+	if jm.Status == "Pushed" || jm.Status == "Already exists" {
```
Comment (williamdes) [inline pkg/compose/push.go:147]: can this strings not use some sort of enum or const ?
(I don't know golang)

**Review by williamdes — COMMENTED** [review]
I tested it with my projects that had the initial scolling bug of docker-compose
Using artifact from: https://github.com/docker/compose/actions/runs/4182986999

The new UI you propose is comprehensive and clean !
It's easy to understand

Very good work !

File: pkg/compose/push.go (hunk 1 above)
Comment (ndeloof) [reply pkg/compose/push.go:147]: unfortunately golang doesn't have enums
ideally, const should be defined by the Docker Engine API

**Comment by williamdes** [comment]
let me know if the recent changes need that I retry
this is a good test project https://github.com/wdesportes/phpmyadmin-local-setup/blob/main/docker-compose.yml

**Review by StefanScherer — APPROVED** [review]
I tried it on Windows in PowerShell/CMD/Windows terminal and looks good there as well.

**Pull request merged**